spec/typedecl_spec.lua

```diff
@@ -8,4 +8,10 @@ describe("Typedecl", function()
             typedecl.declare(mod, "TESTTYPE", "Foo", { Bar = {"x"} })
         end, "tag name 'TESTTYPE.Foo.Bar' is already being used")
     end)
+
+    it("'match_tag' works as expected.", function ()
+        assert.falsy(typedecl.match_tag("foo.Bar.baz", "f.o.Bar"))
+        assert.truthy(typedecl.match_tag("foo.Bar.baz", "foo.Bar"))
+        assert.falsy(typedecl.match_tag("types.T.Float", "types.T."))
+    end, "")
```

Comment:
* We could put each test case in a separate `it` block. That way we can give a descriptive name for each of them. For example the first test can say that it is testing that the "." is not being interpreted as a regex character.
* The second test case could test that the return value is `"baz"` instead of only testing that it is truthy
* I think we don't need to put that `"")` at the end

Reply:
True, should have gone with more descriptive test logs.
Done :+1:.